src/map/clif.cpp

```diff
@@ -24901,6 +24901,53 @@ void clif_macro_reporter_status(map_session_data &sd, e_macro_report_status styp
 #endif
 }
 
+void clif_goldpc_info( struct map_session_data& sd ){
+#if PACKETVER_MAIN_NUM >= 20140508 || PACKETVER_RE_NUM >= 20140508 || defined(PACKETVER_ZERO)
+	if( battle_config.feature_goldpc_active ){
+		struct PACKET_ZC_GOLDPCCAFE_POINT p = {};
+
+		p.packetType = HEADER_ZC_GOLDPCCAFE_POINT;
+		p.active = true;
+		if( battle_config.feature_goldpc_vip && pc_isvip( &sd ) ){
+			p.unitPoint = 2;
+		}else{
+			p.unitPoint = 1;
+		}
+		p.point = (int32)pc_readreg2( &sd, GOLDPC_POINT_VAR );
+		if( sd.goldpc_tid != INVALID_TIMER ){
+			p.accumulatePlaySecond = (int32)( 3600 - battle_config.feature_goldpc_time + pc_readreg2( &sd, GOLDPC_SECONDS_VAR ) );
```

Comment:
I think it's better to change this to:

		if( sd.goldpc_tid != INVALID_TIMER ){
			const struct TimerData* td = get_timer( sd.goldpc_tid );

			if( td != nullptr ){
				// Get the remaining milliseconds until the next reward
				t_tick remaining = td->tick - gettick();

				// Always round up to full second
				remaining += ( remaining % 1000 );

				p.accumulatePlaySecond = (int32)( 3600 - remaining / 1000 );
			}
		}else{
			p.accumulatePlaySecond = 3600;
		}

and the client does not update the counter when changing maps or jumping, which loses synchronization with the server.

Sorry for my English I use a translator.

Reply:
should be added:

	if( battle_config.feature_goldpc_active ){
		clif_goldpc_info( *sd );
	}

at the end of 'clif_parse_LoadEndAck'